text/0079-cdk-2.0.md

```diff
@@ -0,0 +1,632 @@
+changes require a v2 release to signal to users that code changes may be required for adoption.
+
+The main change of note is to consolidate all of the aws service construct libraries into a single package. The
+motivation for this change is detailed in [the corresponding RFC](https://github.com/aws/aws-cdk-rfcs/issues/6).
+
+The core motivation of this proposal is to detail the plan required to release this change while continuing to support
+users referencing the v1 modules. In short, we want to make sure users have plenty of time to transition their code to
+using the new module structure and make that transition as easy as possible.
+
+# Scope
+
+Since this is the first major version release since we announced General Availability of the AWS CDK, we will pay close
+attention to which changes are scoped in for this release. The main criteria used to determine whether a change is
+suitable for inclusion are:
+
+- the change _requires_ breaking existing customers of the changed API or feature
+  - a migration should be possible without incurring replacement of existing resources
+- the change addresses a well documented customer pain point
+- the change has had the due diligence done:
+  - implications of the change are well documented
+  - an implementation strategy is documented
+
+## Changes considered for v2 inclusion
+
+- The following changes will be included, or are candidates for inclusion:
+
+  | Decision | RFC / Issue | Description                                                                         |
+  | -------- | ----------- | ----------------------------------------------------------------------------------- |
+  | ✅ In    |             | Remove `@deprecated` APIs from the previous major version                           |
+  | ✅ In    | [RFC-55]    | Reset all feature flags from the previous major version                             |
+  | ✅ In    | [RFC-6]     | Combine all AWS Construct Libraries into a single package                           |
+  | ✅ In    | [RFC-192]   | Remove the _constructs compatibility layer_ in favor of using `constructs` directly |
+
+* The following changes were scoped out because a solution is likely to be achievable without breaking existing
+  customers:
+
+  | Decision | RFC / Issue | Description                                                                |
+  | -------- | ----------- | -------------------------------------------------------------------------- |
+  | 🚫 Out   | [#39]       | Distribute lambda code for bundled custom resources via a public S3 bucket |
+  | 🚫 Out   | [RFC-77]    | Generate CloudFormation Resources from the CloudFormation Registry schemas |
+  | 🚫 Out   | [RFC-193]   | Fix API for TypeScript type unions when used in statically typed languages |
+  | 🚫 Out   | [#116]      | Easier identification of `@experimental` modules and APIs                  |
```

Comment:
What is the plan for "experimental" modules and APIs in CDK v2.0? Too many customers have come to the wrong conclusion about the stability of the CDK because they were using experimental modules and didn't know it; presumably because they didn't read the API reference labels. The note above this table states the solution is "likely" achievable without breaking customers, but how we solve this particular problem is not stated. I like the proposal in issue #116 to have 2 versions of the CDK; one stable and one stable+experimental.

Reply:
The non-breaking thing we can do here quickly is add `"(experimental)"` to the front of every experimental API's docstring.

Two different libs won't work well as soon as we involve 3rd party libraries.